src/main/java/com/epam/grid/engine/provider/job/slurm/SlurmJobProvider.java

```diff
@@ -263,6 +318,42 @@ private Listing<Job> mapToJobListing(final List<String> stdOut) {
     }
 
     private boolean jobNotFoundByIdError(final CommandResult result) {
-        return result.getStdErr().toString().equals(jobIdNotFoundMessage);
+        if (!result.getStdErr().isEmpty()) {
+            return result.getStdErr().get(0).equals(jobIdNotFoundMessage);
+        }
+        return false;
+    }
+
+    /**
+     * This method creates the structure of the executed job deletion command based on the passed options.
+     *
+     * @param filter User-defined conditions for searching for deleted jobs.
+     * @return The structure of an executable command.
+     */
+    private String[] makeScancelCommand(final DeleteJobFilter filter) {
+        final Context context = new Context();
+        context.setVariable(JOB_FILTER, filter);
+        return commandCompiler.compileCommand(getProviderType(), SCANCEL_COMMAND, context);
+    }
+
+    private List<Long> parseDeletedJobId(final List<String> stdOut) {
```

Comment:
better to rename `stdOut` to something less specific (`outputLines` maybe or `output`)

Reply:
done